navigator/runtime-fragment/src/main/java/com/freeletics/mad/navigator/fragment/NavEventNavigationHandler.kt

```diff
@@ -105,68 +88,13 @@ public open class NavEventNavigationHandler : NavigationHandler<FragmentNavEvent
-                val options = NavOptions.Builder()
-                    .setPopUpTo(event.popUpToDestinationId, inclusive = event.inclusive)
-                    .build()
-                controller.navigate(event.route.destinationId, event.route.getArguments(), options)
-            }
-            is NavEvent.NavigateToRootEvent -> {
-                val controller = fragment.findNavController()
-                val options = NavOptions.Builder()
-                    // save the state of the current root before leaving it
-                    .setPopUpTo(controller.graph.startDestinationId, inclusive = false, saveState = true)
-                    // restoring the state of the target root
-                    .setRestoreState(event.restoreRootState)
-                    // makes sure that if the destination is already on the backstack, it and
-                    // everything above it gets removed
-                    .setLaunchSingleTop(true)
-                    .build()
-                controller.navigate(event.root.destinationId, event.root.getArguments(), options)
-            }
-            is UpEvent -> {
-                val controller = fragment.findNavController()
-                controller.navigateUp()
-            }
-            is BackEvent -> {
-                val controller = fragment.findNavController()
-                controller.popBackStack()
-            }
-            is BackToEvent -> {
-                val controller = fragment.findNavController()
-                controller.popBackStack(event.destinationId, event.inclusive)
-            }
-            is ActivityResultEvent<*> -> {
-                val request = event.request
-                val launcher = activityLaunchers[request] ?: throw IllegalStateException(
-                    "No launcher registered for $request!\nMake sure you called the appropriate " +
-                        "AbstractNavigator.registerFor... method"
-                )
-                @Suppress("UNCHECKED_CAST")
-                (launcher as ActivityResultLauncher<Any?>).launch(event.input)
-            }
-            is NavEvent.PermissionsResultEvent -> {
-                val request = event.request
-                val launcher = permissionLaunchers[request] ?: throw IllegalStateException(
-                    "No launcher registered for $request!\nMake sure you called the appropriate " +
-                        "AbstractNavigator.registerFor... method"
-                )
-                @Suppress("UNCHECKED_CAST")
-                (launcher as ActivityResultLauncher<Any?>).launch(event.permissions)
-            }
-            is FragmentResultEvent -> {
-                val result = Bundle(1).apply {
-                    putParcelable(KEY_FRAGMENT_RESULT, event.result)
-                }
-                fragment.parentFragmentManager.setFragmentResult(event.requestKey, result)
+        if (event is FragmentResultEvent) {
```

Comment:
Should fields of `FragmentResultEvent` be marked as internal too?

Reply:
I have a follow up that splits out `FragmentResultEvent` into it's own thing that doesn't extend `NavEvent` anymore. That change also makes the fields internal.